### ISSUE TITLE
Alert triggers shell command

### ISSUE THREAD

**Issue by randombugs** [TwinProduction/gatus#118]
New Issue as discussed in #117 .

Having the option to execute commands would also allow for 'custom alerts' like sending alerts through other Services then implemented. An Example is Signal Messenger which needs signal-cli for sending messages.

But the possibilities are endless... :)

TwinProduction Reply https://github.com/TwinProduction/gatus/issues/117#issuecomment-850726894

> As for the ability to execute custom commands, that's a little tricky, because Gatus' Docker image uses the scratch image, which means that adding support for any custom commands would require also modifying the Docker image.
>
> I'm not saying it's off the table, but for now, unless somebody has an actual use for it, I don't think it's worth considering.

**Comment by TwiN** [comment]
#47 will likely need to be completed before we can do this, since the Docker image built uses scratch and has no shell of its own.

**Comment by Sim0nW0lf** [comment]
Any news here?

My usecase would be to just try to restart the program which would be really good! In most cases that will resolve the issue in my setups...

**Comment by TwiN** [comment]
> My usecase would be to just try to restart the program which would be really good! 
@Sim0nW0lf You can _technically_ already do by deploying an app with an API that allows you to restart your application(s) and calling it via the `custom` alerting provider.

**Comment by ser** [comment]
I would second shell alert trigger, I potentially could setup API triggering mqtt, but it's easier to call binary doing that in one line.

**Comment by TwiN** [comment]
External endpoints allow you to do this without compromising Gatus' container image now, making this feature request obsolete. Closing.
